pkg/apiserver/openapi/openapi.go

```diff
@@ -29,7 +29,21 @@ import (
 
 var verbs = util.CreateTrie([]string{"get", "log", "read", "replace", "patch", "delete", "deletecollection", "watch", "connect", "proxy", "list", "create", "patch"})
 
-// ToValidOperationID makes an string a valid op ID (e.g. removing punctuations and whitespaces and make it camel case)
+// forcePrefixForOpIDs contains list of operation IDs that will get a prefix (usually GroupVersion) to prevent duplicates.
+var forcePrefixForOpIDs = map[string]bool{
+	"getAPIResources": true,
+	"getAPIGroup":     true,
+	"getAPIVersions":  true,
+}
+
+// forcePrefixForTags contains list of tags which their Operation IDs will get a prefix (usually GroupVersion) to prevent duplicates.
+// It is preferred to put alpha/beta (e.g., *_v1beta1) tags in this list rather than released tags (e.g. CoreV1).
+// Tags are usually lowercase group or group_version
+var forcePrefixForTags = map[string]bool{
```

Comment:
@ncdc @smarterclayton Thanks for looking into this. I appreciate it if we can get this PR in 1.5. 

The point of ToValidOperationID method is to customize OperaionIDs and in my opinion any API server should customize its own. Almost all of the code in that method is specific to the way we setup k8s API server. Things like api->apis/core, apis prefix, removing .k8s.io suffix, fixing Scale OpID) are all k8s specific. That is why I suggest we move this function to master. ToValidOperationID can be reused and I can move it to a common package.